Use installed third-party dependencies instead of Git submodules.

GitHub-generated source archives do not include the contents of the spdlog,
cpu_features, and GoogleTest submodules. Find the corresponding installed
CMake packages and link against their exported targets.

Author: Pavel Tomanek (Inuits/UGent) with help from ChatGPT5.6
--- orig/CMakeLists.txt	2026-08-05 10:22:04.559156598 +0000
+++ new/CMakeLists.txt	2026-08-05 10:22:05.850102211 +0000
@@ -105,7 +105,7 @@
 endif()
 
 find_package(Threads)
-add_subdirectory(third_party/spdlog EXCLUDE_FROM_ALL)
+find_package(spdlog CONFIG REQUIRED)
 
 set(PRIVATE_INCLUDE_DIRECTORIES
   ${CMAKE_CURRENT_SOURCE_DIR}/src
@@ -283,12 +283,8 @@
   if(BUILD_SHARED_LIBS)
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
   endif()
-  set(BUILD_SHARED_LIBS_SAVED "${BUILD_SHARED_LIBS}")
-  set(BUILD_SHARED_LIBS OFF)
-  set(BUILD_TESTING OFF)
-  add_subdirectory(third_party/cpu_features EXCLUDE_FROM_ALL)
-  set(BUILD_SHARED_LIBS "${BUILD_SHARED_LIBS_SAVED}")
-  list(APPEND LIBRARIES cpu_features)
+  find_package(CpuFeatures CONFIG REQUIRED)
+  list(APPEND LIBRARIES CpuFeatures::cpu_features)
 endif()
 
 if(ENABLE_CPU_DISPATCH)
--- orig/tests/CMakeLists.txt	2026-08-05 10:22:04.566719694 +0000
+++ new/tests/CMakeLists.txt	2026-08-05 10:22:05.850365573 +0000
@@ -1,8 +1,6 @@
 enable_testing()
 
-option(BUILD_GMOCK "" OFF)
-option(INSTALL_GTEST "" OFF)
-add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../third_party/googletest ${CMAKE_CURRENT_BINARY_DIR}/googletest)
+find_package(GTest CONFIG REQUIRED)
 
 add_executable(ctranslate2_test
   attention_test.cc
@@ -21,7 +19,7 @@
   )
 target_link_libraries(ctranslate2_test PRIVATE
   ${PROJECT_NAME}
-  gtest_main
+  GTest::gtest_main
   )
 
 add_executable(benchmark_ops
